src/Metadata/ApiResource.php

```diff
@@ -147,8 +148,15 @@ public function __construct(
         protected ?array $graphQlOperations = null,
         $provider = null,
         $processor = null,
-        protected array $extraProperties = []
+        protected array $extraProperties = [],
+        protected ?PersistenceMeansInterface $persistenceMeans = null,
     ) {
+        if (null !== $this->elasticsearch) {
+            $solution = $this->elasticsearch
+                ? sprintf('Pass an instance of %s to $persistenceMeans instead', ElasticsearchDocument::class)
+                : 'You will have to remove it when upgrading to v4';
+            trigger_deprecation('api-platform/core', '3.1', sprintf('Setting "elasticsearch" is deprecated. %s', $solution));
+        }
```

Comment:
Please move this deprecation to the deprecated factory.

Reply:
https://github.com/api-platform/core/pull/5297#discussion_r1060352779